tests/test_api.py

```diff
@@ -383,6 +391,10 @@ def test_non_existing_uuid(self):
                              username=None,
                              uuid='FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF')
 
+        # Check if there are no transactions created when there is an error
+        transactions = Transaction.objects.all()
+        self.assertEqual(len(transactions), 2)
+
```

Comment:
This is not clear. There are already transactinos there but the number seems to be random. I think we should check this using the creation date, so you can check how many transactions were created after the API call. The same for the rest of the tests.